src/bootstrap/builder.rs

```diff
@@ -1554,7 +1554,7 @@ impl<'a> Builder<'a> {
                 // flesh out rpath support more fully in the future.
                 rustflags.arg("-Zosx-rpath-install-name");
                 Some("-Wl,-rpath,@loader_path/../lib")
-            } else if !target.contains("windows") {
+            } else if !target.contains("windows") && !target.contains("aix") {
```

Comment:
You will have to put librustc_driver.so and libstd.so in bin/ instead of lib/ for AIX too then, right?

Reply:
It's very unfortunate that AIX doesn't support rpath and `$ORIGIN` directive.

> does AIX not look for dynamic libraries in the same directory as the executable?

It doesn't.

We are currently using launch scripts to wrap executables. To be more specific, in tree
```
.
├── bin
│   ├── .cargo
│   ├── cargo
│   ├── .rustc
│   └── rustc
└── lib
    ├── librustc_driver.a
    └── libtest.a
```
`bin/cargo` and `bin/rustc` are wrappers, where we implement something like
```shell
EXEC_DIR=`/usr/bin/dirname $0`
EXEC=`/usr/bin/basename $0`
BASE=`/usr/lib/instl/realpath "${EXEC_DIR}/../"`
export LIBPATH="${BASE}/lib":"${LIBPATH:-/usr/lib:/lib}"
exec "${EXEC_DIR}/.${EXEC}" "$@"
```
I guess using rustup to generate `rustc` and `cargo` can also mitigate this issue.